src/vm/ilmarshalers.h

```diff
@@ -599,14 +599,21 @@ class ILMarshaler
                 nativeSize = wNativeSize;
             }
 
-#if defined(_TARGET_X86_)
+#if defined(_TARGET_X86_) || (defined(_TARGET_AMD64_) && defined(_WIN32))
             // JIT32 and JIT64 (which is only used on the Windows Desktop CLR) has a problem generating
             // code for the pinvoke ILStubs which do a return using a struct type.  Therefore, we
             // change the signature of calli to return void and make the return buffer as first argument. 
 
             // for X86 and AMD64-Windows we bash the return type from struct to U1, U2, U4 or U8
             // and use byrefNativeReturn for all other structs.
             // for UNIX_X86_ABI, we always need a return buffer argument for any size of structs.
+#if defined(_TARGET_AMD64_) && defined(_WIN32)
```

Comment:
Why to limit this to AMD64 & WIN32 only? The problem exists on ARM or Unix too, I think.

Reply:
I don't believe that this problem exists on ARM or Unix. It's a weird corner case of the Windows x86/x64 calling conventions for member functions. And the SystemV ABI has more complex rules for return values that have to be dealt with in the SystemV classifier/the JIT (which I believe is already implemented there). And as far as I can tell from a test I did on Compiler Explorer, I think Windows x64 is the only 64-bit platform to pass a hidden buffer argument: https://godbolt.org/z/owtpm8. I can't check Windows x64 ARM though so I'll try to test that locally.

Reply:
I've verified that ARM32 doesn't have this problem.